app/controllers/onboarding_controller.rb

```diff
@@ -0,0 +1,34 @@
+class OnboardingController < ApplicationController
+  before_action :check_activation_status
+
+  layout "unauthenticated"
+
+  def show
+    @user = current_user
+  end
+
+  def update
+    @user = current_user
+
+    @user.skip_password_validation = true
```

Comment:
Bunu sanki modelde kontrol etmeliyiz? password validation create anında kontrol edilmeli sonrasında ise password değiştirilmek istendiğinde değil mi? Ayrıca bu tarz kullanımlar için rails validation context diye bakarsanız bir yöntem görebilirsiniz bu yaptığınıza benzeyecek şekilde :)